src/provider/providerWrapper.ts

```diff
@@ -206,6 +217,47 @@ export class ProviderWrapper {
     }
   }
 
+  private async _getAllDataBatch(
+    address: string,
+    keyHashes: string[],
+  ): Promise<GetDataReturn[]> {
+    if (this.type === ProviderTypes.ETHEREUM) {
+      const encodedResults = await this.callContract(
+        constructJSONRPC(
+          address,
+          Method.GET_DATA_BATCH,
+          abiCoder.encodeParameter('bytes32[]', keyHashes),
+        ),
+      );
+
+      const decodedValues = decodeResult(Method.GET_DATA_BATCH, encodedResults);
+
+      return keyHashes.map<GetDataReturn>((keyHash, index) => ({
+        key: keyHash,
+        value: decodedValues ? decodedValues[index] : decodedValues,
+      }));
+    }
+
+    const payload: JsonRpc[] = [
+      constructJSONRPC(
+        address,
+        Method.GET_DATA_BATCH,
+        abiCoder.encodeParameter('bytes32[]', keyHashes),
+      ),
+    ];
+
+    const results: any = await this.callContract(payload);
+    const decodedValues = decodeResult(
+      Method.GET_DATA_BATCH,
+      results[0].result,
+    );
+
+    return keyHashes.map<GetDataReturn>((key, index) => ({
+      key,
+      value: decodedValues ? decodedValues[index] : decodedValues,
+    }));
+  }
+
```

Comment:
I would reco to not create this funciton which is 95% the same as the function below.

I bet there were a few copy pastes here.

<img width="1407" alt="image" src="https://github.com/ERC725Alliance/erc725.js/assets/477945/b93641b9-8cca-4fbe-b926-5748a9002bbc">

More code more problems, and please DRY ;)
It makes it hard to see what is the diff between batch and non batch, i used the diff checker to see that the only difference is that you use `Method.GET_DATA_BATCH` instead of `Method.GET_DATA`.

So, you add 40 lines of code to just replace 3 variables.

I suggest:

```
getData(address: string, keyHashes: string, useBatchMethod: boolean) {
 // ...
 const decodedValues = decodeResult(
      useBatchMethod ? Method.GET_DATA_BATCH : Method.GET_DATA,
      results[0].result,
    );
// ...
}
```

And you can also add comment here to see that getData was replaced by getDataBatch from version 5